### PR TITLE
6c information gathering

### PR DESCRIPTION
**Pull request by su-vikas**
Issue #1376 

Added content for Information Gathering using Frida for an iOS application.

### CODE REVIEW

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 1]
```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
 
 To learn more, please refer to the [r2frida wiki](https://github.com/enovella/r2frida-wiki/blob/master/README.md "r2frida Wiki").
 
+##### Information Gathering
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]: Can you move this section to line 716 instead? We should describe first how to get the names of classes/methods before we are describing "Method hooking".

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 2]
```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
...
+
+In previous section we learned about using radare2 to collect runtime information of a running process. Radare2 is internally using Frida (r2frida) to obtain process information. In this section we will learn how to use Frida directly to gather process related information.
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]: When moving this part, please remove the first sentence.

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 3]
```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
...
+
+##### Getting Loaded Classes and their Methods
+
+In Frida REPL Objective-C runtime related information can be accessed using `ObjC` command. Within `ObjC` command, `enumerateLoadedClasses` function lists the loaded classes for a given application.
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]:
```suggestion
In the Frida REPL Objective-C runtime the `ObjC` command can be used to access information within the running app. Within the `ObjC` command the function `enumerateLoadedClasses` lists the loaded classes for a given application.
```

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 4]
```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
...
+
+```
+$ frida -U -f com.iOweAss
+
+[iPhone::com.iOweAss]-> ObjC.enumerateLoadedClasses()
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]: Which app is this, iOweAss?

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 5]
```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
...
+    "- coreLogic",
+    "- bat",
+    "- b:",
+    "- hexString:"
+]
+```
+
+##### Getting Loaded Libraries
+
+In Frida REPL process related information ca be obtained using `Process` command. Within `Process` command, `enumerateModules` function lists the libraries loaded into process memory.
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]:
```suggestion
In Frida REPL process related information can be obtained using the `Process` command. Within the `Process` command the function `enumerateModules` lists the libraries loaded into the process memory.
```

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 6]
```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
...
+Similarly, information related to various threads can be obtained.
+
+```
+Process.enumerateThreads()
+[
+    {
+        "context": {
+            "fp": "0x16fd72b90",
+            "lr": "0x1a17a7aa0",
+            "pc": "0x1a17a8634",
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]: Can you please remove all lines from here till "x9"? To make the output more readable.
Just replace it with "..."

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 7]
```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
...
+            "x9": "0x7000906"
+        },
+        "id": 1287,
+        "state": "waiting"
+    },
+
+    ...
+```
+
+`Process` command exposes multiple functions which can be explored as per needs, some useful functions being `findModuleByAddress`, `findModuleByName`, `enumerateRanges` etc.
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]:
```suggestion
The `Process` command exposes multiple functions which can be explored as per needs. Some useful functions are `findModuleByAddress`, `findModuleByName` and `enumerateRanges` besides others.
```

File: Document/0x06c-Reverse-Engineering-and-Tampering.md (hunk 4 above)
Comment (su-vikas) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: It is a crackme app I [wrote](https://github.com/su-vikas/crackmes/tree/master/ios), and was present on my device while writing this. The uncrackable level 1 dont have enough classes to show this example. Is it fine?

File: Document/0x06c-Reverse-Engineering-and-Tampering.md (hunk 6 above)
Comment (su-vikas) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: Done

**Comment by su-vikas** [comment]
@sushi2k I have fixed things as per your suggestions. Please review the changes :)

File: Document/0x06c-Reverse-Engineering-and-Tampering.md (hunk 4 above)
Comment (sushi2k) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: Got it. Can you please remove the "Ass" :-)

**Comment by sushi2k** [comment]
See my other comment about removing "Ass", then we are fine from my side :-)

File: Document/0x06c-Reverse-Engineering-and-Tampering.md (hunk 4 above)
Comment (sushi2k) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: Is this app publicly available?

Comment (su-vikas) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: haha sure. Yes, I linked above, it is here:  https://github.com/su-vikas/crackmes

Comment (su-vikas) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: Fixed!

**Pull request merged**